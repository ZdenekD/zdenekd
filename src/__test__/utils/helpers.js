import React from 'react';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';

/**
 * Factory function to create a shallow wrapper for App component
 * @function setup
 * @param {ReactComponent} Component - React component to render with shallow
 * @param {Object} props - Component props specific to this setup
 * @param {Object} state - Initial state for setup
 * @returns {ShallowWrapper}
 */
export const factory = (Component, props = {}, state = null) => {
    const wrapper = shallow(<Component {...props} />);

    if (state) {
        wrapper.setState(state);
    }

    return wrapper;
};

/**
 * Return ShallowWrapper containg node(s) with the data-test value
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} value - data attribute value to find
 * @returns {ShallowWrapper}
 */
export const findComponent = (wrapper, value) => wrapper.find(`[data-test="${value}"]`);

/**
 * Check corret properties in component
 * @param {ReactComponent} component
 * @param {object} props
 * @returns {Error|undefined}
 */
export const checkProps = (component, props) => checkPropTypes(component.propTypes, props, 'props', component.name);
