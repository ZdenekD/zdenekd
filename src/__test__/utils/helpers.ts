import {ShallowWrapper} from 'enzyme';

const findComponent = (wrapper: ShallowWrapper, attribute: string): ShallowWrapper => wrapper.find(`[data-test="${attribute}"]`);

export default findComponent;
