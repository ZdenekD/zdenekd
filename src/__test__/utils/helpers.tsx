import {
    mount,
    ReactWrapper,
    ShallowWrapper
} from 'enzyme';
import {Provider} from '../../state';

type IWrapper = ShallowWrapper | ReactWrapper;

const findComponent = (wrapper: IWrapper, attribute: string): IWrapper => wrapper.find(`[data-test="${attribute}"]`);

export const stateFactory = (children: React.ReactNode): IWrapper => {
    const wrapper = mount(
        <Provider>
            {children}
        </Provider>
    );

    return wrapper;
};

export default findComponent;
