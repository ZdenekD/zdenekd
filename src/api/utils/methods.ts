import {IMethods, IOptions} from '@/api/types';

const headers: HeadersInit = {
    Accept: 'application/json',
    'Content-Type': 'application/json; charset=utf-8',
};
const getHeaders = (options?: IOptions): HeadersInit => (
    options?.token
        ? {
            ...headers,
            authorization: `Bearer ${options.token}`,
            'Content-language': options?.language || 'cs',
        }
        : {
            ...headers,
            'Content-language': options?.language || 'cs',
        }
);
const methods: IMethods = {
    get: options => ({
        method: 'GET',
        headers: getHeaders(options),
        signal: options?.signal,
    }),
    post: (data, options) => ({
        method: 'POST',
        headers: getHeaders(options),
        signal: options?.signal,
        body: data ? JSON.stringify(data) : null,
    }),
    put: (data, options) => ({
        method: 'PUT',
        headers: getHeaders(options),
        signal: options?.signal,
        body: JSON.stringify(data),
    }),
    patch: (data, options) => ({
        method: 'PATCH',
        headers: getHeaders(options),
        signal: options?.signal,
        body: JSON.stringify(data),
    }),
    delete: options => ({
        method: 'DELETE',
        headers: getHeaders(options),
        signal: options?.signal,
    }),
};

export default methods;
