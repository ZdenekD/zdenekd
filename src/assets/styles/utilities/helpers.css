:global ._visuallyhidden {
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    position: absolute;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    border: none;
}

:global ._align-right {
    text-align: right !important;
}
