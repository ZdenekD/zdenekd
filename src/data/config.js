const config = {cloudfront: 'https://d2v7x5yqza2kb5.cloudfront.net'};

export default config;
