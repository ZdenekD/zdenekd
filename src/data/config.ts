export interface IConfig {
    cloudfront: string | undefined
}

const config: IConfig = {cloudfront: process.env.NODE_ENV === 'production' ? process.env.NEXT_PUBLIC_CLOUDFRONT : ''};

export default config;
