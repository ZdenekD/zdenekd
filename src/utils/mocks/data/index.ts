import MockDataParagraph from './content/UI/paragraph';

export {MockDataParagraph};
