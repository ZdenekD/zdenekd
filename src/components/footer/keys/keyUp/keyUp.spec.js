import {factoryState, findComponent} from '../../../../__test__/utils/helpers';
import KeyUp from './index';
import data from '../../../../data/pages';

describe('Key Up', () => {
    const defaultProps = {data, router: {route: '/'}};
    let wrapper;

    beforeEach(() => {
        wrapper = factoryState(KeyUp, defaultProps);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-key');

        expect(component.exists()).toBe(true);
    });
});
