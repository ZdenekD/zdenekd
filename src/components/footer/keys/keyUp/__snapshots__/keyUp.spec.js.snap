// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key Up match snapshot 1`] = `
<button class="undefined undefined"
        data-test="component-key"
        type="button"
        aria-label="Předchozí stránka"
        tabindex="-1"
>
  <i>
  </i>
</button>
`;
